tools/src/main/java/ee/jakarta/tck/data/tools/annp/RespositoryProcessor.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package ee.jakarta.tck.data.tools.annp;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.annotation.processing.SupportedOptions;
+import javax.annotation.processing.SupportedSourceVersion;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import ee.jakarta.tck.data.tools.qbyn.QueryByNameInfo;
+import jakarta.data.repository.Repository;
+import jakarta.persistence.Entity;
+
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Types;
+
+
+/**
+ * Annotation processor for {@link Repository} annotations that creates sub-interfaces for repositories
+ * that use Query By Name (QBN) methods.
+ */
+@SupportedAnnotationTypes("jakarta.data.repository.Repository")
+@SupportedSourceVersion(SourceVersion.RELEASE_17)
+@SupportedOptions({"debug", "generatedSourcesDirectory"})
+public class RespositoryProcessor extends AbstractProcessor {
+    private Map<String, RepositoryInfo> repoInfoMap = new HashMap<>();
+
+    @Override
+    public synchronized void init(ProcessingEnvironment processingEnv) {
+        super.init(processingEnv);
+        processingEnv.getOptions();
+    }
+
+    @Override
+    public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+        System.out.printf("RespositoryProcessor: Processing repositories, over=%s\n", roundEnv.processingOver());
+        boolean newRepos = false;
+        Set<? extends Element> repositories = roundEnv.getElementsAnnotatedWith(Repository.class);
+        for (Element repository : repositories) {
+            String provider = repository.getAnnotation(Repository.class).provider();
+            if(provider.isEmpty() || provider.equalsIgnoreCase("hibernate")) {
```

Comment:
> I hope everyone understands that this is something essentially any implementor of Jakarta Data can use to add support for QbMN.

Would this restrict the the use case of this tool to hibernate only?

I'd be interested in a README file to help understand the general use case of this tool for a Jakarta Data provider.

Reply:
Ah yes, that should be parameterized I guess.